packages/clay-data-provider/src/__tests__/LRUCache.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {LRUCache} from '../LRUCache';
+
+let cache: LRUCache<string, {}>;
+
+describe('LRUCache', () => {
+	beforeEach(() => {
+		cache = new LRUCache(3);
+	});
+
+	afterEach(() => {
+		cache.reset();
+	});
+
+	it('throw error if initialized with maxSize 0', () => {
+		expect(() => new LRUCache(0)).toThrow(/non-negative number/);
+	});
+
+	it('throw error if initialized with a negative maxSize', () => {
+		expect(() => new LRUCache(-10)).toThrow(/non-negative number/);
+	});
+
+	it('returns new storage when cache is reset', () => {
+		expect(cache.reset().size).toBe(0);
+	});
+
+	it('returns undefined for undefined key', () => {
+		expect(cache.get('foo')).toBe(undefined);
+	});
+
+	it('returns value for present value', () => {
+		cache.set('foo', 'f');
+		expect(cache.get('foo')).toBe('f');
+	});
+
+	it('least recently set', () => {
```

Comment:
Can you change this into a sentence that reads correct when read out loud? ie. "it something something something". I don't actually know what you are testing here, or in the next one.

Compare that with [these tests I wrote here](https://github.com/wincent/masochist/blob/8ee6fc3704b204adae6dc3ad9baca5108a741637/src/server/__tests__/LRUCache-test.js#L41-L58), where the `it` and the comments describe what is actually expected and happening:

```javascript
describe('set()', () => {
  it('evicts old items', () => {
    // Populate the cache.
    cache.set('a', '#a');
    cache.set('b', '#b');
    cache.set('c', '#c');
    expect(cache.has('a')).toBe(true);
    expect(cache.has('b')).toBe(true);
    expect(cache.has('c')).toBe(true);

    // Cause an eviction.
    cache.set('d', '#d');
    expect(cache.has('a')).toBe(false);
    expect(cache.has('b')).toBe(true);
    expect(cache.has('c')).toBe(true);
    expect(cache.has('d')).toBe(true);
  });
});
```